libcontainer/setns_init_linux.go

```diff
@@ -35,7 +35,10 @@ func (l *linuxSetnsInit) Init() error {
 
 	if !l.config.Config.NoNewKeyring {
 		if err := label.SetKeyLabel(l.config.ProcessLabel); err != nil {
-			return err
+			// workaround for RHEL7 kernel which may return EACCES for an empty label
+			if !(l.config.ProcessLabel == "" && os.IsPermission(err)) {
```

Comment:
I think `if l.config.ProcessLabel != "" || !os.IsPermission(err)` reads nicer.

Reply:
To me your version is less readable, despite looking better. I read mine as "ignore the error unless label is empty and error is permission".

Reply:
@cyphar but I can certainly change it, please let me know

Reply:
I don't really mind either way. I read mine as "this is an error if the label is not empty, or the error is not a permission-related one" but either works.